### PR TITLE
build nodejs with OpenSSL and ICU provided as proper dependencies

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

Changelog:
* build nodejs with external OpenSSL to easily get security fixes
* enable full ICU support, fixes issue https://github.com/jupyterlab/jupyterlab/issues/10487
* reduce duplication in `postinstallcmds`
* enhance sanity checks by also checking the added symlinks

### CODE REVIEW

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

File: easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -2,7 +2,7 @@ easyblock = 'ConfigureMake'
 
 name = 'nodejs'
 version = '14.17.0'  # LTS on 2021-06-03
-local_libversion = '83'
+_libversion = '83'
```
Comment (verdurin) [inline easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb:5]: @lexming why this change?

File: easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb [hunk 2]
```diff
@@ -2,7 +2,7 @@ easyblock = 'ConfigureMake'
 
 name = 'nodejs'
 version = '16.15.1'  # LTS on 2022-06-12
-local_libversion = '93'
+_libversion = '93'
```
Comment (verdurin) [inline easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb:5]: @lexming also here - why the change?

File: easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb:5]: I systematically remove `local` from custom variables in easyconfigs since the change allowing `_variable` as name. It's a better format.

File: easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/n/nodejs/nodejs-16.15.1-GCCcore-11.3.0.eb:5]: I systematically remove `local` from custom variables in easyconfigs since the change allowing `_variable` as name. It's a better format.

File: easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb:5]: I think we're pretty consistent in using `local_*` for local variables in easyconfigs, so we should stick to that in the central easyconfigs repo, I think... "`It's a better format`" is highly subjective, no?

That said, it's not like we should be looking for reasons to keep PRs open longer either.

So as long as we don't enforce either `_*` or `local_*` in the tests, this shouldn't block a PR from being merged.

Comment (boegel) [reply easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb]: It's not wasted effort, because I think we should actually enforce the use of `local_*` in the easyconfig test suite.

EasyBuild allows using either `local_*` or `_*`, but in the central easyconfigs repo it makes sense to be a bit more strict to keep some consistency across easyconfigs we include. Similar, in some sense, to how we require `use_pip`, and restrict dependency versions in a particular easyconfig generation.

Comment (boegel) [reply easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb]: Some "stats": there's currently 205 easyconfigs that use `_*` for local variables, and 3910 that use `local_*`.

So it's not consistent currently, but making it consistent should definitely go towards using `local_*` everywhere instead of `_*`.

Comment (jfgrimm) [reply easybuild/easyconfigs/n/nodejs/nodejs-14.17.0-GCCcore-10.3.0.eb]: @boegel any ideas on the ratio for recent (new) contributions? From my experience, a lot of the `local_` use is in bumped older ECs

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3106.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/079da90b6f974cc1c454586a83ae5dfc for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node148.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/f2286b97a4c05684869e84c6c94636eb for a full test report.
